main.cpp

```diff
@@ -324,11 +289,88 @@ struct IntType
     IntType& subtract(int rhs );
     IntType& multiply(int rhs );
     IntType& divide(int rhs );
-
+    
+    int pow(const IntType&);
+    int pow(const FloatType&);
+    int pow(const DoubleType&);
+    int pow(int);
+    
+    int powInternal(int* val, int arg);
+    
 private:
     int* value;
+    
 };
 
+float FloatType::powInternal( float* val, float arg)
+{
+    *val = std::pow( *val, arg);
+    return *val;
+}
+float FloatType::pow(float floatExp)
+{
+    return powInternal(value, floatExp);
+}
+float FloatType::pow(const FloatType& floatTypeExp)
+{
+    return powInternal(value, const_cast<FloatType&>(floatTypeExp) );
+}
+float FloatType::pow(const IntType& intTypeExp)
+{
+    int it = const_cast<IntType&>(intTypeExp);
+    return powInternal(value, it );
+}
+float FloatType::pow(const DoubleType& doubleTypeExp)
+{
+    double dt = const_cast<DoubleType&>(doubleTypeExp);
+    return powInternal(value, static_cast<float>(dt) );
+}
+
+double DoubleType::powInternal( double* val, double arg)
+{
+    *val = std::pow( *val, arg);
+    return *val;
+}
+double DoubleType::pow(double doubleExp)
+{
+    return powInternal(value, doubleExp);
+}
+double DoubleType::pow(const FloatType& floatTypeExp)
+{
+    return powInternal(value, static_cast<double>( const_cast<FloatType&>(floatTypeExp) ));
+}
+double DoubleType::pow(const IntType& intTypeExp)
+{
+    int it = const_cast<IntType&>(intTypeExp);
+    return powInternal(value, it );
+}
+double DoubleType::pow(const DoubleType& doubleTypeExp)
+{
+    return powInternal(value, const_cast<DoubleType&>(doubleTypeExp) );
+}
+
+int IntType::powInternal( int* val, int arg)
+{
+    *val = static_cast<int>(std::pow( *val, arg));
+    return *val;
+}
+int IntType::pow(int intExp)
+{
+    return powInternal(value, intExp);
+}
+int IntType::pow(const FloatType& floatTypeExp)
+{
+    return powInternal(value, static_cast<int>( const_cast<FloatType&>(floatTypeExp) ));
+}
+int IntType::pow(const IntType& intTypeExp)
+{
+    return powInternal(value, const_cast<IntType&>(intTypeExp) );
+}
+int IntType::pow(const DoubleType& doubleTypeExp)
+{
+    return powInternal(value, static_cast<int>( const_cast<DoubleType&>(doubleTypeExp) ));
+}
```

Comment:
Do not use `const_cast`.  Find another way to make this work.   
hint: what is missing from your conversion function?
- [ ] revise